src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -301,6 +317,39 @@ private async Task PreRunConditions()
             }
         }
 
+        private bool HostJsonExist()
+        {
+            var hostJsonPath = Path.Combine(Environment.CurrentDirectory, "host.json");
+            return File.Exists(hostJsonPath);
+        }
+
+        private bool BundleConfigurationExists()
+        {
+            var hostJsonPath = Path.Combine(Environment.CurrentDirectory, "host.json");
+            var hostJson = FileSystemHelpers.ReadAllTextFromFile(hostJsonPath);
+            return hostJson.Contains("extensionBundle", StringComparison.OrdinalIgnoreCase);
+        }
+
+        private bool IsPreCompiledFunctionApp()
+        {
+            bool isPrecompiled = false;
+            foreach (var directory in FileSystemHelpers.GetDirectories(Environment.CurrentDirectory))
+            {
+                var functionMetadataFile = Path.Combine(directory, "function.json");
+                if (File.Exists(functionMetadataFile))
+                {
+                    var functionMetadataFileContent = FileSystemHelpers.ReadAllTextFromFile(functionMetadataFile);
+                    var functionMetadata = JsonConvert.DeserializeObject<FunctionMetadata>(functionMetadataFileContent);
+                    isPrecompiled = isPrecompiled || (functionMetadata?.ScriptFile?.EndsWith(".dll") != null && functionMetadata.ScriptFile.EndsWith(".dll"));
```

Comment:
use `GetExtension` similar to https://github.com/Azure/azure-functions-host/blob/aa3d0576d5bc2f73a86bdb4c32af3c97be9b9f14/src/WebJobs.Script/Host/FunctionMetadataProvider.cs#L174